src/sha512.cpp

```diff
@@ -0,0 +1,283 @@
+#include <cstdio>
```

Comment:
it's exactly the same, only with a surrounding `namespace libtorrent`, the reason I made this is to minimize changes and easy the review, but if this get accepted, the very next step is remove the old file